### ISSUE TITLE
Accessibility fails on List component

### ISSUE THREAD

**Issue by AlexSkrypnyk** [civictheme/uikit#78]
The A11y review have identified the following issues with a List component (and, potentially, some sub-components):
- [ ] 1. The filters and options are read by a screenreader left to right, top to bottom (e.g. first row left to right and then second row left to right) - **FAIL**    **From John:** Using a keyboard and screenreader on hover the menu should expand and it should read the sub menu items first before going to the next main navigation item.
- [x] 2. The screen reader announces when a filter dropdown is expanded or collapsed - PASS    
- [ ] 3. The focus is visible on each select box in drop down when navigating using a mouse or keyboard - **FAIL**   
- [ ] 4. The user can tab to each interactive element and interact with the filter using keyboard or mouse - **FAIL**   
- [x] 5. The Enter a keyword textbox should have an associated label - PASS    
- [x] 6. The search button has a role of button and a title attribute stating “Click to search by keyword.” - PASS    
- [x] 7. The text and background must have a colour contrast of 4.5:1. - PASS    
- [x] 8. All user interface elements, as well as their states (focus, hover, select, check, visited/unvisited, expand/collapse and outline) should have a colour contrast of 3:1 - PASS    
- [x] 9. The text is still visible and readable when the line height (line spacing) is at least 1.5 times the font size, letter spacing is at least 0.12 times the font size, and word spacing is at least 0.16 times the font size - PASS    
- [ ] 10. If the contents underneath the filter change when an option is selected, the change should be announced using aria-live=”polite” - **FAIL**

**Comment by AlexSkrypnyk** [comment]
1. Need more clarification from John Cloys
3. Confirmed. The [Group filter](https://uikit.civictheme.io/?path=/story/molecules-group-filter--group-filter) does not allow to navigate with a keyboard. Need to investigate.
4. Confirmed. As above.
10. Need to add a JS script to the Single and Group Filter components (separate scripts) that would allow to update a List component where the filter appears on. 
  10.1 The script should provide own events for when the filter changes. See `collapsible.js` for an example of triggering a custom event. The event name is `ct.single-filter.update` for a Single Filter and `ct.group-filter.update` for a Group Filter component.
  10.2 The "targeting" of the list should be done through a `data` attribute that would be using a CSS query to find the targeted List component. If not provided - the closest list component will be used.
  10.3 Once the target List is found, the script should bind to the custom filter event (`ct.single-filter.update` or `ct.group-filter.update`) and set `aria-live=”polite”` attribute.

**Comment by fionamorrison23** [comment]
Hi @jcloys please see item 1 
1. The filters and options are read by a screenreader left to right, top to bottom (e.g. first row left to right and then second row left to right) - FAIL
Can you please test this and confirm the issues? Thanks.
